ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiLayeredWorldState.java

```diff
@@ -257,17 +258,15 @@ public Stream<StreamableAccount> streamAccounts(final Bytes32 startKeyHash, fina
 
   @Override
   public MutableWorldState copy() {
-    final BonsaiPersistedWorldState bonsaiPersistedWorldState =
-        ((BonsaiPersistedWorldState) archive.getMutable());
-    BonsaiInMemoryWorldStateKeyValueStorage bonsaiInMemoryWorldStateKeyValueStorage =
-        new BonsaiInMemoryWorldStateKeyValueStorage(
-            bonsaiPersistedWorldState.getWorldStateStorage().accountStorage,
-            bonsaiPersistedWorldState.getWorldStateStorage().codeStorage,
-            bonsaiPersistedWorldState.getWorldStateStorage().storageStorage,
-            bonsaiPersistedWorldState.getWorldStateStorage().trieBranchStorage,
-            bonsaiPersistedWorldState.getWorldStateStorage().trieLogStorage,
-            bonsaiPersistedWorldState.getWorldStateStorage().getMaybeFallbackNodeFinder());
-    return new BonsaiInMemoryWorldState(archive, bonsaiInMemoryWorldStateKeyValueStorage);
+    // TODO:  this is a stopgap measure, revisit with
+    // https://github.com/hyperledger/besu/issues/4641
+    // rip off the band-aid.  you know you want to:
```

Comment:
I can't justify creating a layered-in-memory-worldstate when we intend to move to snapshots.  Just rip off the band-aid and do it.

Reply:
in retrospect, we need to have a fallback that does not use snapshots.  I reverted the copy behavior to use inMemory and current persisted worldstate, and rolled persisted state in the copy method.  This is effectively the same behavior as allowing MainnetBlockValidator to roll the persisted state, then get a copy.  It just compartmentalizes it.  I think we need to make snapshots the default behavior in #4664, and we can have this behavior effectively 'feature flagged' by only enabling it if we disable snapshots